src/main.cpp

```diff
@@ -0,0 +1,628 @@
+#include "..\include\TradingEngine\order.h"
+#include "..\include\TradingEngine\limit_order.h"
+#include "..\include\TradingEngine\transactions.h"
+#include "..\include\TradingEngine\order_tree.h"
+#include "..\include\TradingEngine\order_book.h"
+
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <sstream>
+#include <vector>
+#include <algorithm>
+#include <queue>
+#include <mutex>
+#include <thread>
+#include<atomic>
+#include <optional>
+#include <condition_variable>
+#include <future>
+
+
+using std::shared_ptr;
+using std::queue;
+using std::mutex;
+using std::thread;
+using std::string;
+using std::atomic;
+using std::condition_variable;
+using std::vector;
+using std::optional;
+
+
+//////////////////////////////////////////////////////////// Single Thread /////////////////////////////////////////////////////////////////////////////////////////////
+
+/* Function to split incoming_order read from file */
+Order convertOrder(const std::string& input) {
+    std::vector<std::string> parts;
+    std::stringstream ss(input);
+    std::string item;
+
+
+    while (std::getline(ss, item, ',')) {
+        parts.push_back(item);
+    }
+
+    /* Access the individual values */
+    OrderType order_type = (parts[0] == "BUY") ? OrderType::BUY : OrderType::SELL;
+    unsigned int order_id = std::stoi(parts[1]);
+    float quantity = std::stof(parts[2]);
+    float price = std::stof(parts[3]);
+
+    return Order(std::move(order_id), std::move(order_type), std::move(quantity), std::move(price));
+}
+
+int main() {
+
+    /* Open the file to read it */
+    std::fstream file("sample_large_order_10k.txt", std::ios::in);
+
+    /* Create your OrderBook object */
+    OrderBook orderBook;
+
+    /* Total Orders submitted */
+    int total_orders_submitted = 0;
+
+    std::string line;
+
+    /* Start the timer */
+    auto startTime = std::chrono::high_resolution_clock::now();
+
+    while (std::getline(file, line)) {
+
+        //std::cout << line << std::endl;
+
+        /* Read order and convert it */
+        Order order(convertOrder(line));
+
+        /* Process the order through the order book */
+        orderBook.processOrder(std::ref(order));
+        total_orders_submitted += 1;
+    }
+
+    /* Stop the timer */
+    auto endTime = std::chrono::high_resolution_clock::now();
```

Comment:
The time you are calculating here is:
Open file + Read record + convert to order + process order.

This is not reflecting what you want because opening file and reading are very expensive operations.

Fix:
Load the file records into a steam, then read from the stream directly and process. Start the timer after loading into the stream,

Btw, you kept the file open for too long, bad practice.